boa_gc/src/pointers/ephemeron.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{
+    finalizer_safe,
+    internals::EphemeronBox,
+    trace::{Finalize, Trace},
+    Allocator, Gc, GcBox, EPHEMERON_QUEUE,
+};
+use std::cell::Cell;
+use std::ptr::NonNull;
+
+pub struct Ephemeron<K: Trace + ?Sized + 'static, V: Trace + ?Sized + 'static> {
+    inner_ptr: Cell<NonNull<GcBox<EphemeronBox<K, V>>>>,
+}
+
+impl<K: Trace + ?Sized, V: Trace> Ephemeron<K, V> {
+    pub fn new(key: &Gc<K>, value: V) -> Self {
+        Self {
+            inner_ptr: Cell::new(Allocator::new(GcBox::new_weak(EphemeronBox::new(
+                key, value,
+            )))),
+        }
+    }
+}
```

Comment:
Shouldn't this also unroot the `value` of the ephemeron?

Reply:
We don't need to root/unroot an ephemeron, since ephemeron's are essentially viewed as rooted by default (by the weak flag) and are passed to `Collector::mark_ephemeron()`